add reference doc for adding client customizations in TypeSpec (#12758)
Azure/azure-sdk-tools: eng/common/knowledge/customizing-client-tsp.md

  create is MyService.createUser;
}
```

### @clientLocation

**Purpose**: Move operations between clients without restructuring.
**Restrictions**: Cannot be used with `@client` or `@operationGroup` decorators.
**Syntax**: `@clientLocation(target: Interface | Namespace | string, scope?: string)`
**Usage**:

```typespec
// Move to existing client
@@clientLocation(MyService.upload, AdminOperations);

// Move to new client
@@clientLocation(MyService.archive, "ArchiveClient");

// Move to root client
@@clientLocation(MyService.SubClient.health, MyService);

// Move parameter to client initialization
@@clientLocation(MyService.upload.subscriptionId, MyService);
```

### @clientName

**Purpose**: Override generated names for SDK elements. Takes precedence over all other naming mechanisms.
**Important**: Always use PascalCase or camelCase for the rename parameter to make it easier to combine language scopes. SDKs will apply language-specific naming conventions automatically.
**Syntax**: `@clientName(rename: string, scope?: string)`
**Usage**:

```typespec
// Rename Type
@@clientName(PetStore, "PetStoreClient");

// Language-specific names
@@clientName(foo, "pythonicFoo", "python")

> @@clientName(foo, "csharpicFoo", "csharp")

Reviewer: ```suggestion
@@clientName(foo, "CSharpFoo", "csharp")
```

```

### @clientNamespace

**Purpose**: Change the namespace/package of generated types in the client SDK.
**Syntax**: `@clientNamespace(rename: string, scope?: string)`
**Usage**:

```typespec
// Change client namespace
@@clientNamespace(MyService, "MyClient");

// Move model to different namespace
@@clientNamespace(MyService.MyModel, "MyClient.Models")
```

### @clientInitialization

**Purpose**: Add custom parameters to client initialization.

> **Important**: When `@clientInitialization` references a model defined in `client.tsp`, a file-level namespace (e.g. `namespace ClientCustomizations;`) should be added if one does not exist.

Reviewer: ```suggestion
**Important**: When `@clientInitialization` references a model defined in `client.tsp`, add a file-level namespace (e.g. `namespace ClientCustomizations;`) if one does not exist.
```

```

### Scenario 2: Rename for Consistency

```typespec
// Standardize naming across languages
@@clientName(MyService, "MyServiceClient");
@@clientName(getUserInfo, "GetUserInformation");

// Language-specific naming
@@clientName(uploadFile, "UploadFile", "csharp, python");
```

### Scenario 3: Add Client Parameters

```typespec
// Elevate common parameters to client
model MyServiceOptions {
  subscriptionId: string;
  resourceGroup: string;
}

@@clientInitialization(MyService, { parameters: MyServiceOptions });
```

### Scenario 4: Multi-Client Architecture

```typespec
// Separate admin and user operations
@client({ service: MyService, name: "AdminClient" })
interface AdminClient {
  deleteUser is MyService.deleteUser;
  manageRoles is MyService.manageRoles;
}

@client({ service: MyService, name: "UserClient" })
interface UserClient {
  getProfile is MyService.getProfile;
  updateProfile is MyService.updateProfile;
}
```

### Scenario 5: Language-specific clients

```typespec
// Different client names for Java and others
@client({ service: MyService, name: "Foo.MyServiceClient" }, "java")
@client({ service: MyService, name: "MyServiceClient" }, "!java")

> interface MyServiceClient {

Author: Both ways work actually. This way just allows us to not repeat the operations that would otherwise be the same:

Example: https://github.com/Azure/azure-rest-api-specs/blob/a68d44d57aac4fb5a61634c8c9ec474e64e59253/specification/ai/Face/client.tsp#L84-L98
